addons/account/models/account_move.py

```diff
@@ -1903,6 +1909,33 @@ def _apply_cash_rounding(self, diff_balance, diff_amount_currency, cash_rounding
 
         _apply_cash_rounding(self, diff_balance, diff_amount_currency, existing_cash_rounding_line)
 
+    @contextmanager
+    def _sync_unbalanced_lines(self, container):
+        yield
+
+        # Unlink tax lines if all tax tags have been removed.
+        if not container['records'].line_ids.tax_ids:
+            container['records'].line_ids.filtered(lambda line: line.tax_line_id).unlink()
+
+        # Unlink the automatic balancing line, if any, to prevent having multiple ones.
+        container['records'].line_ids.filtered(lambda line: line.name == 'Automatic Balancing Line').unlink()
```

Comment:
Need to translate as well, maybe it would be clearer if you extracted the variable?
```suggestion
        balance_name = _('Automatic Balancing Line')
        container['records'].line_ids.filtered(lambda line: line.name == balance_name).unlink()
```

And then use that variable down below too